### PR TITLE
Introduce nfcore datastructure 

### PR DESCRIPTION
**Pull request by Steffengreiner**
- [X] This comment contains a description of changes (with reason)
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] `CHANGELOG.rst` is updated

**Description of changes**
This PR introduces the datastructure files and folders representing the result of parsing a nf-core pipeline bioinformatical output. 
You can see a valid JSON parsing output in the PR [here](https://github.com/qbicsoftware/data-model-lib/pull/197/files#diff-fac647ce12ce0b1e46c60ea0c372be493d47659461e9285aca30c5d13ff2d8e7)
The final result is generated and stored in an NfCoreExperiment Class/Object akin to the NanoporeOutputExperiment Object/Class. 
Click [here](https://drive.google.com/file/d/1FTlWn70t4y0rj_7cBwA_h2KOiTRhJ8K0/view) to see the ER diagram

**Additional context**
The tests currently fail since it includes the changes to the schema and json output that will be introduced in #197.
The ER diagram will be included in the Github Readme once its finalized

### CODE REVIEW

**Comment by jenniferboedker** [comment]
Reminder: update the image of the folder structures. It does not reflect the inputids and the runids in the result set

File: src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+/**
+ * Class that holds information about a the output of a nf-core pipeline
+ *
+ * @since 2.5.0
+ */
+final class NfCoreExperiment {
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy:16]:
```suggestion
final class NfCorePipelineResult {
```

File: src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+    private final static Set nfCoreFolderTypes = [
+            FQDN_FOLDERS + ".PipelineInformationFolder",
+            FQDN_FOLDERS + ".QualityControlFolder",
+            FQDN_FOLDERS + ".ProcessFolder"
+    ]
+
+    private NfCoreExperiment(List<?> resultSet) {
+        this.resultSet = Objects.requireNonNull(resultSet as Object, "resultSet must not be null") as List<?>
+    }
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy:41]: An nf core output is the result set. And this result set has properties, as described in the ER.

File: src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+        //Add files to resultSet
+        //ToDo How should the properties and ProcessFolders be included in the Experiment File?
+        List<?> resultSet = [[pipelineInformation, pipelineInformationProperties], qualityControl, processFolders, runId, sampleIds]
+        return new NfCoreExperiment(resultSet)
+    }
+
+    List<?> getResultSet() {
+        return this.resultSet
+    }
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy:95]: The object shall provide getters for the individual result set properties

**Review by sven1103 — CHANGES_REQUESTED** [review]
Thank you @Steffengreiner for this contribution.

I think the NfCoreExperiment needs some refactoring. Especially the result sets list was irritating me, as it is not modelled in the ER.

File: src/main/groovy/life/qbic/datamodel/datasets/NfCorePipelineResult.groovy [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+     * @return
+     */
+    String getProcessFolders() {
+        return processFolders
+    }
+
+    /**
+     * Provides access to the information stored in the runId file
+     * @return
+     */
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/NfCorePipelineResult.groovy:146]:
```suggestion
    /**
     * Returns the analysis run identifier provided by Nextflow Tower.
     * @return
     */
```

File: src/main/groovy/life/qbic/datamodel/datasets/NfCorePipelineResult.groovy [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+        return runId
+    }
+
+    /**
+     * Provides access to the information stored in the sampleIds file
+     * @return
+     */
+    String getSampleIds() {
+        return sampleIds
+    }
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/datasets/NfCorePipelineResult.groovy:157]:
```suggestion
    /**
     * Returns a list of QBiC sample identifier that served as input for the analysis run.
     * @return
     */
    List<String> getSampleIds() {
        return sampleIds
    }
```

File: src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy (hunk 2 above)
Comment (Steffengreiner) [reply src/main/groovy/life/qbic/datamodel/datasets/NfCoreExperiment.groovy]: Done!

**Review by sven1103 — APPROVED** [review]
Great contribution @Steffengreiner . I approve it now and will give feedback when I have incorporated it in the Java ETL.

**Pull request merged**